File: src/modules/authentication/services/auth_reset_password_service.py
from src.modules.authentication.domain.dto.auth_reset_password_dto import (
    AuthResetPasswordDTO,
)
from src.modules.user.models import User
from src.modules.user.validators.user_password_validator import user_password_validator
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.services.token_verify_service import TokenVerifyService
from src.shared.utils.hash import hash


class AuthResetPasswordService:
    def execute(dto: AuthResetPasswordDTO):
        payload = TokenVerifyService.execute(token=dto.token)

        user = User.objects.filter(email=payload["sub"], deleted_at=None).first()

        if user is None:
            raise NotFoundException("Token inválido!")

        user_password_validator(dto.password)

        user.password = hash(dto.password)
        user.save()

        return user

File: src/modules/auth/services/auth-reset-password.service.ts
import { Injectable, UnauthorizedException } from '@nestjs/common';
import { hash } from 'bcryptjs';
import { PrismaService } from 'src/infra/database/prisma.service';
import { UserEntity } from 'src/modules/user/domain/entities/user.entity';
import { TokenVerifyService } from 'src/shared/services/token-verify.service';
import { AuthResetPasswordDto } from '../domain/dto/auth-reset-password.dto';

@Injectable()
export class AuthResetPasswordService {
  constructor(
    private prisma: PrismaService,
    private tokenVerifyService: TokenVerifyService,
  ) {}
  async execute(dto: AuthResetPasswordDto): Promise<UserEntity> {
    const user = await this.tokenVerifyService.execute(dto.token);

    if (!user) throw new UnauthorizedException('Token inválido!');

    const updatedUser = await this.prisma.user.update({
      data: {
        password: await hash(dto.password, 10),
      },
      where: { id: user.id },
      include: {
        role: true,
        organization: true,
        company: true,
        avatar: true,
      },
    });

    return new UserEntity(updatedUser);
  }
}
